hbase-client/src/main/java/org/apache/hadoop/hbase/RegionMetrics.java

```diff
@@ -156,8 +156,8 @@ default String getNameAsString() {
   int getStoreRefCount();
 
   /**
-   * @return the max reference count for any store file among all stores files
+   * @return the max reference count for any store file among all compacted stores files
    *   of this region
    */
-  int getMaxStoreFileRefCount();
+  int getMaxCompactedStoreFileRefCount();
```

Comment:
Because `RegionMetrics` is a public interface, could you please deprecate `getMaxStoreFileRefCount` and add `getMaxCompactedStoreFileRefCount` as a new method?

Reply:
Oh okk, yes ideally we would do that but in this case, this change `int getMaxStoreFileRefCount();` has not yet landed to any release so far and is available to 2.3.0 and 1.6.0 only, no other release branch. Hence, the plan is to update this method now only before 2.3.0 or 1.6.0 makes it to releases.
Thanks

Reply:
Ah, ok. This seems like a minor gap in our documentation on how we handle deprecations. Would be good to have a common sense and a documentation for it. @saintstack Any thoughts?

Reply:
Agree with the suggestion. 
This new IA method was introduced as part of Jira [HBASE-22460](https://issues.apache.org/jira/browse/HBASE-22460) and [HBASE-23213](https://issues.apache.org/jira/browse/HBASE-23213) which are not part of any active release so far. Hence IMO, before 2.3.0 or 1.6.0 goes live, ideally we should be good to make this change, but yes agree to follow standard best practices reg such cases.

cc: @apurtell @anoopsjohn